extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleJobStore.java

```diff
@@ -333,4 +339,46 @@ private Key getDataKey(UUID jobId, String key) {
       throw new RuntimeException("Failed to deserialized entityKey: " + entityKey, t);
     }
   }
+
+  @Override
+  public void create(UUID jobId, String key, InputStream stream) throws IOException {
+    Preconditions.checkNotNull(jobId);
+    Transaction transaction = datastore.newTransaction();
+    Key fullKey = getDataKey(jobId, key);
+    Entity shouldNotExist = transaction.get(fullKey);
+    if (shouldNotExist != null) {
+      transaction.rollback();
+      throw new IOException(
+          "Record already exists for key: " + fullKey.getName() + ". Record: " + shouldNotExist);
+    }
+
+    String serialized = objectMapper.writeValueAsString(stream);
+    Entity entity = Entity.newBuilder(fullKey)
+        .set(CREATED_FIELD, Timestamp.now())
+        .set(InputStream.class.getName(), serialized)
+        .build();
+
+    try {
+      transaction.put(entity);
+    } catch (DatastoreException e) {
+      throw new IOException(
+          "Could not create initial record for jobID: " + jobId + ". Record: " + entity, e);
+    }
+    transaction.commit();
+  }
+
+  @Override
+  public InputStream getStream(UUID jobId, String key) throws IOException {
+    Key fullKey = getDataKey(jobId, key);
+    Entity entity = datastore.get(fullKey);
+    if (entity == null) {
+      return null;
+    }
+    try {
+      return objectMapper
+          .readValue(entity.getString(InputStream.class.getName()), InputStream.class);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to deserialize entityKey: " + fullKey, e);
```

Comment:
Since the create() method throws IOException, this one should probably rethrow this exception as well, perhaps wrapped in another IOException with the full key information.